.github/workflows/ci.yml

```diff
@@ -175,9 +185,9 @@ jobs:
       SKIP: "pylint"
       MOUNT_LOCAL_SOURCES: "true"
       PYTHON_MAJOR_MINOR_VERSION: ${{needs.build-info.outputs.defaultPythonVersion}}
-    if: github.repository == 'apache/airflow' || github.event_name != 'schedule'
-      # We want to make sure we have latest sources as only in_container scripts are added
-    # to the image but we want to static-check all of them
+    if: >
+      needs.build-info.outputs.needs-image-build == 'true' &&
```

Comment:
Running (or not running) static checks based on if we need to build an image or not seems confusing. I know this makes sense right now, but it feels like a side-effect: for instance a possible optimization we might do in future once we have more selective tests of providers is to rebuild the image much less often and mount sources in.

Could we come up with a new name for this output, even if it's always set in the same place as needs-image-build right now?

Reply:
Actually. it's a good name in this case (I even had it done like that before). I thought about renaming it, but I think it reflects pretty well both intention and behaviour.

With this change,, we add two versions of static checks - one that runs with CI image built ("Static checks"  and the other without "Statich checks: no CI image"). If (in selective test scripts) we decide that image build is needed - we run the full set of static checks, if we decide that we do not need it, we run the "no-ci-image one". I think it is really proper logic and not a side effect at all.

Reply:
I decided to change the name similar to your suggestion and described the logic now.